### PR TITLE
Review the Java File of the Build Command

### LINKED ISSUE #19220: Typos in error message given for `ballerina build -c`
The error message you get when you use the `-c` flag for a file contains a typo. And can possibly be rephrased in a better way.
```
$ ballerina build -c test2.bal 
ballerina: '-c' or '--compile' flag cannot be used on Ballerina files. the flag can only be used with Ballerina projects.
```
Maybe as follows:
```
ballerina: the '-c' | '--compile' flag can only be used with modules.
```

### PR DESCRIPTION
**Pull request by praneesha**
Fixes #19220

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java [hunk 1]
```diff
@@ -68,8 +68,8 @@
  *
  * @since 0.90
  */
-@CommandLine.Command(name = BUILD_COMMAND, description = "Ballerina build - Builds Ballerina module(s) and generates " +
-                                                         "executable outputs.")
+@CommandLine.Command(name = BUILD_COMMAND, description = "Ballerina build - Build Ballerina module(s) and generate " +
+                                                         "executable output.")
```
Comment (pubudu91) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java]: This is sorta wrong, even the previous message. The binary built by the `build` command is not necessarily executable in the sense that it doesn't have to have an entry point.

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java [hunk 2]
```diff
@@ -116,26 +116,26 @@ public BuildCommand(Path userDir, PrintStream outStream, PrintStream errStream,
     @CommandLine.Option(names = {"--all", "-a"}, description = "Build or compile all the modules of the project.")
     private boolean buildAll;
     
-    @CommandLine.Option(names = {"--output", "-o"}, description = "Writes output to the given file. The provided " +
-                                                                  "output filename may or may not contain the '.jar' " +
+    @CommandLine.Option(names = {"--output", "-o"}, description = "Write the output to the given file. The provided " +
```
Comment (pubudu91) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java]: "Writes" is correct in this context isn't it? This is basically saying "The build commands writest the output..."

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java [hunk 3]
```diff
@@ -166,51 +166,51 @@ public void execute() {
...
             }
         
-            //// validate and set source root path
+            //// Validate and set the path of the source root.
             if (!ProjectDirs.isProject(this.sourceRootPath)) {
                 Path findRoot = ProjectDirs.findProjectRoot(this.sourceRootPath);
                 if (null == findRoot) {
                     CommandUtil.printError(this.errStream,
-                                           "you are trying to build/compile a Ballerina project but there is no " +
+                                           "You are trying to build/compile a Ballerina project that does not have a " +
```
Comment (pubudu91) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java]: This should be "you", going by the convention we use

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java [hunk 4]
```diff
@@ -224,17 +224,15 @@ public void execute() {
             targetPath = this.sourceRootPath.resolve(ProjectDirConstants.TARGET_DIR_NAME);
 
             if (args.length > 0) {
-                CommandUtil.printError(this.errStream, "too many arguments.", buildCmd, false);
+                CommandUtil.printError(this.errStream, "Too many arguments.", buildCmd, false);
```
Comment (pubudu91) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java]: Again, "too" is correct here.

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java [hunk 5]
```diff
@@ -294,7 +292,7 @@ public void execute() {
             //// check if command executed from project root.
             if (!RepoUtils.isBallerinaProject(this.sourceRootPath)) {
                 CommandUtil.printError(this.errStream,
-                        "you are trying to build/compile a module that is not inside a project.",
+                        "You are trying to build/compile a module that is not inside a project.",
```
Comment (pubudu91) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java]: You -> you

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java [hunk 6]
```diff
@@ -304,8 +302,8 @@ public void execute() {
             //// check if module name given is not absolute.
             if (Paths.get(argList.get(0)).isAbsolute()) {
                 CommandUtil.printError(this.errStream,
-                                       "you are trying to build/compile a module by giving the absolute path. you " +
-                                               "only need give the name of the module.",
+                                       "You are trying to build/compile a module by giving the absolute path. You " +
```
Comment (pubudu91) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java]: You -> you

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java [hunk 7]
```diff
@@ -334,10 +332,10 @@ public void execute() {
             targetPath = this.sourceRootPath.resolve(ProjectDirConstants.TARGET_DIR_NAME);
         } else {
             CommandUtil.printError(this.errStream,
-                                   "invalid Ballerina source path. It should either be a name of a module in a " +
+                                   "Invalid Ballerina source path. It should either be a name of a module in a " +
```
Comment (pubudu91) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java]: Invalid -> invalid

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java (hunk 1 above)
Comment (pubudu91) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java]: As per the offline discussion we had with Kishanthan, shall we mention in the help text of build that executables are produced only for modules containing an entry point?

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java (hunk 2 above)
Comment (praneesha) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java]: We use "write" as a convention following the practice used in docs of similar communities like [1].

[1] https://golang.org/cmd/go/

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java (hunk 3 above)
Comment (praneesha) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java]: fixed.

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java (hunk 4 above)
Comment (praneesha) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java]: Fixed.

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java (hunk 5 above)
Comment (praneesha) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java]: Fixed.

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java (hunk 6 above)
Comment (praneesha) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java]: Fixed.

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java (hunk 7 above)
Comment (praneesha) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java]: Fixed.

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java [hunk 8]
```diff
@@ -270,7 +268,7 @@ public void execute() {
                 try {
                     targetPath = Files.createTempDirectory("ballerina-build-" + System.nanoTime());
                 } catch (IOException e) {
-                    throw LauncherUtils.createLauncherException("error occurred when creating executable.");
+                    throw LauncherUtils.createLauncherException("Error occurred when creating executable.");
```
Comment (MaryamZi) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java]: Shouldn't this be 
```suggestion
                    throw LauncherUtils.createLauncherException("Error occurred when creating the executable.");
```

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java [hunk 9]
```diff
@@ -304,8 +302,8 @@ public void execute() {
             //// check if module name given is not absolute.
             if (Paths.get(argList.get(0)).isAbsolute()) {
                 CommandUtil.printError(this.errStream,
-                                       "you are trying to build/compile a module by giving the absolute path. you " +
-                                               "only need give the name of the module.",
+                                       "you are trying to build/compile a module by giving the absolute path. You " +
```
Comment (MaryamZi) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java]:
```suggestion
                                       "you are trying to build/compile a module giving the absolute path. You " +
```

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java [hunk 10]
```diff
@@ -406,13 +404,13 @@ public String getName() {
 
     @Override
     public void printLongDesc(StringBuilder out) {
-        out.append("Builds Ballerina module(s)/file and produces an executable jar file(s). \n");
+        out.append("Build Ballerina module(s)/file and produce an executable JAR file(s). \n");
```
Comment (MaryamZi) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java]:
```suggestion
        out.append("Build a Ballerina module(s)/file and produce an executable JAR file(s). \n");
```

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java (hunk 1 above)
Comment (praneesha) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java]: Fixed in the cli/ballerina-tool/src/main/resources/cli-help/ballerina-build.help file.

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java [hunk 11]
```diff
@@ -166,51 +166,51 @@ public void execute() {
...
         
-        // when -a or --all flag is provided. check if the command is executed within a ballerina project. update source
-        // root path if command executed inside a project.
+        // When -a or --all is provided, check if the command is executed within a Ballerina project. Update source
+        // root path if the command is executed inside a project.
         if (this.buildAll) {
-            // when building all the modules
-            //// check if output flag is set
+            // Check if the output flag is set
+            //// when building all the modules.
```
[the remark this answers is not in the document]
Comment (praneesha) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java]: Added the comment as a single line;

 // Check if the output flag is set when building all the modules.

**Pull request merged**